### PR TITLE
Add autofill v2 pixels

### PR DESCRIPTION
**Pull request by THISISDINOSAUR**
Task/Issue URL: https://app.asana.com/0/0/1202830175334806/f
Tech Design URL:
CC:

**Description**:
Also adds the alert when attempting to create a duplicate
New ones are:
m_autofill_logins_settings_enabled
m_autofill_logins_settings_disabled

m_autofill_logins_settings_add-new-login_error_attempted-to-create-dupliplicate

m_autofill_logins_save_disable-prompt_shown
m_autofill_logins_save_disable-prompt_autofill-kept
m_autofill_logins_save_disable-prompt_autofill-disabled

Changed pixels
m_secure-vault_is-enabled-checked_when-enabled-and-backgrounded
Changed to only be sent when the app is backgrounded

**Steps to test this PR**:
1. Test the above pixels are fired appropriately (e.g. using Charles Proxy)
2. Test that attempting to create a duplicate login fires the alert 
3. Test that creating a non duplicate login does not fire the alert
4. Do the same as the last two steps, but when editing a login

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `'`

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 14
* [ ] iOS 15
* [ ] iOS 16

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**

### CODE REVIEW

File: Core/PixelEvent.swift [hunk 1]
```diff
@@ -556,15 +563,22 @@ extension Pixel.Event {
...
             return "m_autofill_logins_autoprompt_dismissed"
             
+        case .autofillLoginsFillLoginInlineDisablePromptShown: return "m_autofill_logins_save_disable-prompt_shown"
+        case .autofillLoginsFillLoginInlineDisablePromptAutofillKept: return "m_autofill_logins_save_disable-prompt_autofill-kept"
+        case .autofillLoginsFillLoginInlineDisablePromptAutofillDisabled: return "m_autofill_logins_save_disable-prompt_autofill-disabled"
+            
         case .autofillSettingsOpened: return "m_autofill_settings_opened"
+        case .autofillLoginsSettingsEnabled: return "m_autofill_logins_settings_enabled"
+        case .autofillLoginsSettingsDisabled: return "m_autofill_logins_settings_disabled"
+        case .autofillLoginsSettingsAddNewLoginErrorAttemptedToCreateDuplicate: return "m_autofill_logins_settings_add-new-login_error_attempted-to-create-dupliplicate"
```
Comment (amddg44) [inline Core/PixelEvent.swift]: Heads up, this is throwing a linter line length violation. Also, `dupliplicate` looks like a typo (although it is the same as in Asana 🙂 )

File: DuckDuckGo/AutofillLoginDetailsViewModel.swift [hunk 2]
```diff
@@ -175,16 +176,18 @@ final class AutofillLoginDetailsViewModel: ObservableObject {
...
-            switch viewMode {
-            case .edit:
-                guard let accountID = account?.id else {
-                    assertionFailure("Trying to save edited account, but the account doesn't exist")
-                    return
-                }
-                
-                let vault = try SecureVaultFactory.default.makeVault(errorReporter: SecureVaultErrorReporter.shared)
-                
+        guard let vault = try? SecureVaultFactory.default.makeVault(errorReporter: SecureVaultErrorReporter.shared) else {
```
Comment (amddg44) [inline DuckDuckGo/AutofillLoginDetailsViewModel.swift:180]: Another linter issue

File: DuckDuckGo/TabViewController.swift [hunk 3]
```diff
@@ -2454,8 +2454,8 @@ extension TabViewController: SecureVaultManagerDelegate {
         let isEnabled = featureFlagger.isFeatureOn(.autofill)
         let isBackgrounded = UIApplication.shared.applicationState == .background
         let pixelParams = [PixelParameters.isBackgrounded: isBackgrounded ? "true" : "false"]
```
Comment (amddg44) [inline DuckDuckGo/TabViewController.swift]: Just a nit, `let pixelParams = [PixelParameters.isBackgrounded: isBackgrounded ? "true" : "false"]` can be moved into the if condition

**Review by amddg44 — CHANGES_REQUESTED** [review]
All of the new pixels fired correctly for me, just a couple of minor tweaks needed 👍

File: Core/PixelEvent.swift (hunk 1 above)
Comment (THISISDINOSAUR) [reply Core/PixelEvent.swift]: Good spot, done!

**Review by amddg44 — APPROVED** [review]
LGTM!

**Pull request merged**